### PR TITLE
Move load_yaml_file to common

### PR DESCRIPTION
**Pull request by kit1980**
This is for https://github.com/pytorch/pytorch/pull/131724 and future timm_models.py refactoring.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames

### CODE REVIEW

**Review by shunting314 — APPROVED** [review]

File: benchmarks/dynamo/common.py [hunk 1]
```diff
@@ -284,6 +285,36 @@ class CI(NamedTuple):
 }
 
 
+@functools.lru_cache(maxsize=1)
```
Comment (ysiraichi) [inline benchmarks/dynamo/common.py:288]: Should we increase this cache max size? i.e. is there any case where we call this function twice with different filenames?

Comment (kit1980) [reply benchmarks/dynamo/common.py:288]: "call this function twice with different filenames" - that's unlikely and in any way is not really a problem - the old value will be evicted from the cache.

Comment (huydhn) [reply benchmarks/dynamo/common.py:288]: Maybe we can just keep `maxsize=None` here and be done with it?

**Review by huydhn — APPROVED** [review]

File: benchmarks/dynamo/common.py (hunk 1 above)
Comment (kit1980) [reply benchmarks/dynamo/common.py:288]: I think `maxsize=1` is fine and reflects the intended usage.

**Pull request closed without merging**